src/map/map.cpp

```diff
@@ -194,6 +194,507 @@ struct s_generator_options {
 } gen_options;
 #endif
 
+const std::string MapZoneDatabase::getDefaultLocation() {
+	return std::string(db_path) + "/map_zones.yml";
+}
+
+/**
+ * Reads and parses an entry from map_zones.
+ * @param node: YAML node containing the entry.
+ * @return count of successfully parsed rows
+ */
+uint64 MapZoneDatabase::parseBodyNode(const ryml::NodeRef& node) {
+	std::string zone_name;
+
+	if (!this->asString(node, "Id", zone_name))
+		return 0;
+
+	std::string zone_constant = "MAPTYPE_" + zone_name;
+	int64 zone_id_const;
+
+	if (!script_get_constant(zone_constant.c_str(), &zone_id_const)) {
+		this->invalidWarning(node["Id"], "Zone %s is not valid.\n", zone_name.c_str());
+		return 0;
+	}
+
+	uint16 zone_id = static_cast<uint16>(zone_id_const);
+	std::shared_ptr<c_map_zone> zone = this->find(zone_id);
+	bool exists = zone != nullptr;
+
+	if (!exists) {
+		zone = std::make_shared<c_map_zone>();
+		zone->id = zone_id;
+	}
+
+	if (this->nodeExists(node, "DisabledCommands")) {
+		const auto &commandNode = node["DisabledCommands"];
+
+		for (const auto &it : commandNode) {
+			std::string command_name;
+			c4::from_chars(it.key(), &command_name);
+
+			if (!atcommand_exists(command_name.c_str())) {
+				this->invalidWarning(commandNode, "Atcommand %s does not exist.\n", command_name.c_str());
+				continue;
+			}
+
+			uint16 group_lv;
+
+			if (!this->asUInt16(commandNode, command_name, group_lv))
+				continue;
+
+			if (group_lv > 100) {
+				this->invalidWarning(commandNode, "Atcommand %s's Group Level can not be above 100, capping to 100.\n", command_name.c_str());
+				group_lv = 100;
+			}
+
+			if (group_lv == 0)
+				zone->disabled_commands.erase(command_name);
+			else
+				zone->disabled_commands[command_name] = group_lv;
+		}
+	}
+
+	if (this->nodeExists(node, "DisabledSkills")) {
+		const auto &skillNode = node["DisabledSkills"];
+
+		for (const auto &it : skillNode) {
+			std::string skill_name;
+
+			if (!this->asString(it, "Skill", skill_name))
+				return 0;
+
+			uint16 skill_id = skill_name2id(skill_name.c_str());
+
+			if (skill_id == 0) {
+				this->invalidWarning(skillNode, "Skill %s does not exist.\n", skill_name.c_str());
+				continue;
+			}
+
+			for (const auto &subBl : it) {
+				std::string bl_name;
+				c4::from_chars(subBl.key(), &bl_name);
+
+				if (bl_name.compare("Skill") == 0)
```

Comment:
Just because i had to think a little bit to understand "why" this check is here:
Would it be better to remove this string check (that may be case sensitive) and add a BL_SKILL check below the BL_NUL check instead?